### PR TITLE
Add Composite Solver

### PR DESCRIPTION
**Pull request by lucas-maes**

### CODE REVIEW

File: docs/api/solver.md [hunk 1]
```diff
@@ -1,9 +1,199 @@
...
+actions = out["actions"]   # (n_envs, horizon, action_dim)
+```
+
+## **[ Design Philosophy ]**
+
+The package is built around four ideas. They are worth reading once
+before subclassing or composing solvers — most subtleties downstream
+follow from them.
+
+### 1. One outer loop, four hooks
```
Comment (quentinll) [inline docs/api/solver.md:31]: 4 or 5?

**Comment by keon** [comment]
I found one issue while checking the new `CompositeSolver` path: when a continuous child uses `CEMSolver`, the returned action can violate the configured `Box` action-space bounds.

This appears to be a pre-existing `CEMSolver` behavior surfaced by the composite setup, rather than a bug in `CompositeSolver` itself. But it matters for `CompositeSolver` because a mixed `Dict` action space may return a dict where the discrete component is valid while the continuous component is outside its subspace bounds.

### Evidence

With `Box(low=-1, high=1, ...)`, `CEMSolver` can return actions around `4–5` when the model cost favors an out-of-bounds optimum:

```text
CEMSolver:
min = 4.387
max = 4.965
in_bounds = False
```

For comparison, the same fixture with `ICEMSolver` stays in bounds:

```text
ICEMSolver:
min = 0.996
max = 0.999
in_bounds = True
```

### Repro

```py
import numpy as np
import torch
from gymnasium import spaces as gym_spaces

from stable_worldmodel.policy import PlanConfig
from stable_worldmodel.solver.cem import CEMSolver
from stable_worldmodel.solver.icem import ICEMSolver


class FavorsOutOfBounds:
    # Minimum is at +5, outside Box[-1, 1].
    # If the solver respects action_space, final actions should stay <= 1.
    def get_cost(self, info, candidates):
        return ((candidates - 5.0) ** 2).sum(dim=(-1, -2))


def run_solver(cls):
    kwargs = dict(
        model=FavorsOutOfBounds(),
        n_steps=8,
        num_samples=128,
        topk=16,
        seed=0,
    )
    if cls.__name__ == "CEMSolver":
        kwargs["var_scale"] = 2.0

    solver = cls(**kwargs)
    space = gym_spaces.Box(low=-1, high=1, shape=(2, 2), dtype=np.float32)

    solver.configure(
        action_space=space,
        n_envs=2,
        config=PlanConfig(horizon=3, receding_horizon=1, action_block=1),
    )

    out = solver({"state": torch.zeros(2, 1)})
    a = out["actions"]

    print(cls.__name__)
    print("min", float(a.min()))
    print("max", float(a.max()))
    print("in_bounds", bool((a >= -1).all() and (a <= 1).all()))


run_solver(CEMSolver)
run_solver(ICEMSolver)
```
